test/reports/encoding_reports/test_DesignMatrixExporter.py

```diff
@@ -48,3 +49,34 @@ def test_generate(self):
         self.assertEqual("test_encoding", loaded["encoding"])
 
         shutil.rmtree(path)
+
+    def test_exporter(self):
+        dataset = RepertoireDataset(encoded_data=EncodedData(examples=csr_matrix(np.arange(12).reshape(3, 4)),
+                                                             labels={"l1": [1, 0, 1], "l2": [0, 0, 1]},
+                                                             example_ids=[0, 1, 2],
+                                                             feature_names=["f1", "f2", "f3", "f4"],
+                                                             encoding="test_encoding"))
+
+        path = EnvironmentSettings.tmp_test_path / "designmatrrixexporterreport/"
+
+        report = DesignMatrixExporter(dataset=dataset, result_path=path,
+                                      name="design_matrix", format_file='.csv')
+        report.generate_report()
+        self.assertTrue(os.path.isfile(path / "design_matrix.csv"))
+        report.format = 'csv.zip'
+        report._export_matrix()
+        self.assertTrue(os.path.isfile(path / "design_matrix.csv.zip"))
+
+        report.format = 'npy'
+        report._export_matrix()
+        self.assertTrue(os.path.isfile(path / "design_matrix.npy"))
+        report.format = 'npy.zip'
+        report._export_matrix()
+        self.assertTrue(os.path.isfile(path / "design_matrix.npy.zip"))
+
+        report.format = 'hdf5'
+        report._export_matrix()
+        self.assertTrue(os.path.isfile(path / "design_matrix.hdf5"))
+        report.format = 'hdf5.zip'
+        report._export_matrix()
+        self.assertTrue(os.path.isfile(path / "design_matrix.hdf5.zip"))
```

Comment:
minor thing: could you remove the test directory after the test is finished? There is an example in the previous function :)

Reply:
done :)